pkg/virt-operator/install-strategy/create.go

```diff
@@ -1511,6 +1511,58 @@ func addOrRemoveSSC(targetStrategy *InstallStrategy,
 	return nil
 }
 
+func syncKubevirtNamespaceLabels(kv *v1.KubeVirt, stores util.Stores, clientset kubecli.KubevirtClient) error {
+
+	targetNamespace := kv.ObjectMeta.Namespace
+	obj, exists, err := stores.NamespaceCache.GetByKey(targetNamespace)
+	if err != nil {
+		log.Log.Errorf("Failed to retrieve kubevirt namespace from store. Error: %s", err.Error())
+		return err
+	}
+	if !exists {
+		return fmt.Errorf("Could not find namespace in store. Namespace key: %s", targetNamespace)
+	}
+
+	cachedNamespace := obj.(*corev1.Namespace)
+	// Prepare namespace metadata patch
+	targetLabels := map[string]string{
+		"openshift.io/cluster-monitoring": "true",
+	}
+	cachedLabels := cachedNamespace.ObjectMeta.Labels
+	labelsToPatch := make(map[string]string)
+	for targetLabelKey, targetLabelValue := range targetLabels {
+		cachedLabelValue, ok := cachedLabels[targetLabelKey]
+		if ok && cachedLabelValue == targetLabelValue {
+			continue
+		}
+		labelsToPatch[targetLabelKey] = targetLabelValue
+	}
+
+	if len(labelsToPatch) == 0 {
+		log.Log.Infof("Kubevirt namespace (%s) labels are in sync", targetNamespace)
+		return nil
+	}
+
+	labelsPatch, err := json.Marshal(labelsToPatch)
+	if err != nil {
+		log.Log.Errorf("Failed to marshal namespace labels: %s", err.Error())
+		return err
+	}
+
+	log.Log.Infof("Patching namespace %s with %s", targetNamespace, labelsPatch)
+	_, err = clientset.CoreV1().Namespaces().Patch(
+		targetNamespace,
+		types.MergePatchType,
```

Comment:
I think that this will replace all labels just with the target labels: https://kubernetes.io/docs/tasks/run-application/update-api-object-kubectl-patch/#use-a-json-merge-patch-to-update-a-deployment. The only sane way to patch is to either use strategic merge patch (if that is supported the right way on labels) or using a json patch with `test` and `replace`.

Reply:
@rmohr Can you please elaborate on how can I determine whether a resource supports strategic merge patch vs test+replace?

Reply:
test/replace is always supported. Here an example where we use it: https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-controller/watch/vmi.go#L393

Reply:
for strategic merge, probably best to just try it out ... Otherwise the openapi spec should also contain hints.

Reply:
but I am not really sure. Maybe the merge patch works exactly like we want on maps. Just check.

Reply:
@rmohr JSON merge patch works in this case because the labels are a dictionary. If we were patching arrays we would need a strategic merge patch instead.